lib/Settings.js

```diff
@@ -61,9 +63,33 @@ Settings.loadRepositoriesSettings = function (configPath) {
         }, repositorySettings).then(function (settings) {
             repositories[name] = new RepositorySettings(settings);
         });
-    });
+    }).then(setupGoogleAPI);
 };
 
+/** This function reads the Google API config and initializes the client library.
+ * It writes the configuration in a GoogleConfig.json, because the Google client expects
+ * the config to be in a file.
+ *
+ * @return {Promise<Object>} Returns an instance to the Google Sheets API client.
+ */
+function setupGoogleAPI() {
+    if (!defined(Settings.googleApiConfig) || !defined(Settings.individualClaSheetID) || !defined(Settings.individualClaSheetID)) {
+        return;
+    }
+
+    var googleConfigFilePath = 'GoogleConfig.json';
+    fs.writeFileSync(googleConfigFilePath, JSON.stringify(Settings.googleApiConfig));
+
+    return googleapis.google.auth.getClient({
+        keyFile: googleConfigFilePath,
+        scopes: ['https://www.googleapis.com/auth/spreadsheets']
+    }).then(function(client) {
```

Comment:
Our convention these days is to use async/await for everything, will make the code much cleaner.